cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------
+# Auto creation of CMake targets from pkg-config
+#
+# Copyright (C) 2017 Alexandre Pretyman. All rights reserved.
+#
+# Looks for a pkgconfig module of a given name and exports a cmake
+# target of the same name
+# ----------------------------------------------------------------------
+
+include(FindPkgConfig)
+
+include(hunter_internal_error)
+include(hunter_status_debug)
+
+function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
+  if(TARGET "${PKG_CONFIG_MODULE}")
+    return()
+  endif()
+  pkg_check_modules(${PKG_CONFIG_MODULE} ${PKG_CONFIG_MODULE})
```

Comment:
Yea, link lines are above, but for clarity, `IMPORTED_TARGET`:

```
/usr/bin/gcc   -rdynamic CMakeFiles/x264-example.dir/example.c.o  -o x264-example
```

Hunter's version:
```
/usr/bin/gcc   -rdynamic CMakeFiles/x264-example.dir/example.c.o  -o x264-example -L/home/discipline/Development/CMake/hunter/_Base/xxxxxxx/bd0d80f/ddaf47f/Install/lib -lx264 -lpthread -lm -ldl
```